File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/ReportsResource.java
package work.socialhub.kmastodon.api;

import mastodon4j.entity.Report;
import mastodon4j.entity.share.Response;

/**
 * @author hecateball
 */
public interface ReportsResource {

    /**
     * Fetching a user's reports.
     *
     * @return a list of Reports made by the authenticated user
     */
    public Response<Report[]> getReports();

    /**
     * Reporting a user.
     *
     * @param accountId
     * @param statusIds
     * @param comment
     * @return the finished Report
     */
    public Response<Report> postReport(String accountId, String[] statusIds, String comment);
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/ReportsResource.kt
package work.socialhub.kmastodon.api

import work.socialhub.kmastodon.api.request.reports.ReportsPostReportRequest
import work.socialhub.kmastodon.api.response.Response
import work.socialhub.kmastodon.api.response.reports.ReportsPostReportResponse
import work.socialhub.kmastodon.api.response.reports.ReportsReportsResponse
import kotlin.js.JsExport

@JsExport
interface ReportsResource {

    /**
     * Fetching a user's reports.
     */
    suspend fun reports(
    ): Response<Array<ReportsReportsResponse>>

    @JsExport.Ignore
    fun reportsBlocking(
    ): Response<Array<ReportsReportsResponse>>

    /**
     * Reporting a user.
     */
    suspend fun postReport(
        request: ReportsPostReportRequest
    ): Response<ReportsPostReportResponse>

    @JsExport.Ignore
    fun postReportBlocking(
        request: ReportsPostReportRequest
    ): Response<ReportsPostReportResponse>
}
